### PR TITLE
Api return all positions

### PR DESCRIPTION
**Pull request by yhua045**
The api should return all events for the requested period even when there are no scheduled events at all.

### CODE REVIEW

**Comment by Josephu** [comment]
It really would be good to have an integration test (using supertest, as I had in the `/hello` endpoint), to see the exact changes of the response. I do find it hard to follow what exactly happens, so I will feel much more confident if I can see a test running over it. Maybe a quick walk through for me also helps.

**Comment by yhua045** [comment]
Ok, I have added some integration tests. I hope that helps making it a bit clearer :)

**Comment by Josephu** [comment]
@yhua045 Thanks. I have added a bit more detail to the integration test, mainly to visualize the data payload to ensure what really has happened. I can see now we load all positions for each calendar date regardless if event exist or not. Approve.

Thanks you for your work to write the integration tests!

File: api/service/events-service.js [hunk 1]
```diff
@@ -30,6 +34,51 @@ class EventService {
...
+  }
+  static linkScheduledEventsToCalendarDates(dateRange, scheduledEvents) {
+    return PositionRepository.getAll().then(positions => {
+      const dates = EventService.getWeekdayDatesForTimePeriod(dateRange);
+      let allEvents = EventService.linkPositionsToDates(positions, dates);
+
+      scheduledEvents.forEach(scheduledEvent => {
+        let event = allEvents.find(event => {
+          return (
+            moment(event.calendarDate.date).format(dateTimeFormat.stringFormat) ==
```
Comment (Josephu) [inline api/service/events-service.js]: you should always use `===` instead of `==`

File: api/service/events-service.js [hunk 2]
```diff
@@ -30,6 +34,51 @@ class EventService {
...
+              moment(scheduledEvent.calendarDate.date).format(dateTimeFormat.stringFormat)
+              && event.position.name == scheduledEvent.position.name
+          );
+        });
+        if(event == null){
```
Comment (Josephu) [inline api/service/events-service.js]: I think when no event is found, it is actually `undefined`. You can get away from `==` as it will catch both `undefined` and `null`. I am fine with this way, but you have to be mindful of this behaviour.

File: api/service/events-service.js [hunk 3]
```diff
@@ -30,6 +34,51 @@ class EventService {
...
+
+    return days;
+  }
+  static linkScheduledEventsToCalendarDates(dateRange, scheduledEvents) {
+    return PositionRepository.getAll().then(positions => {
+      const dates = EventService.getWeekdayDatesForTimePeriod(dateRange);
+      let allEvents = EventService.linkPositionsToDates(positions, dates);
+
+      scheduledEvents.forEach(scheduledEvent => {
+        let event = allEvents.find(event => {
```
Comment (Josephu) [inline api/service/events-service.js]: 好hard core的寫法啊 ... 全部掃過一個一個比。

File: api/service/events-service.js (hunk 1 above)
Comment (yhua045) [reply api/service/events-service.js]: thanks will fix this.

File: api/service/events-service.js (hunk 2 above)
Comment (yhua045) [reply api/service/events-service.js]: great, thanks for the information.

File: api/service/events-service.js (hunk 3 above)
Comment (yhua045) [reply api/service/events-service.js]: 我實在想不出來其他的辦法 你有沒有好的建議 不需要現在改 但如果有比較好的方式 我還是希望改掉這寫法

Comment (Josephu) [reply api/service/events-service.js]: 我不是很確定scheduledEvent的資料結構，也許 https://lodash.com/docs/#find 派得上用場，可以簡化程式？不過你的case要比較 momentJS 物件，可能不會work。

File: test/integration-test/server-test.js [hunk 4]
```diff
@@ -1,20 +1,88 @@
...
+    it('returns the 3 weeks of data between 2017-10-01 and 2017-10-15', function() {
+      return request(app)
+        .get('/api/events?from=2017-10-01&to=2017-10-15&mock=false')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .then(function(res) {
+          expect(res.body.data).to.eql([
+            {
+              date: '2017-10-15',
+              members: [
```
Comment (Josephu) [inline test/integration-test/server-test.js]: 我覺得整個資料結構印出來，會比較確定你真的做的是如想像的。我相信本來就對了，不過seeing is believing 😸

File: api/service/events-service.js (hunk 3 above)
Comment (yhua045) [reply api/service/events-service.js]: 好 謝了 我會看一看 是不是有什麼可以使用的

File: test/integration-test/server-test.js (hunk 4 above)
Comment (yhua045) [reply test/integration-test/server-test.js]: 👍  我同意 這個好 畢竟這連日期也一起側了 我就是懶惰了
